File: test_gemini.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  const apiKey = '';
  final url = Uri.parse('https://generativelanguage.googleapis.com/v1beta/models/gemini-2.0-flash:generateContent?key=$apiKey');

  final body = jsonEncode({
    'contents': [
      {
        'parts': [
          {'text': 'Hello! Are you working?'}
        ]
      }
    ]
  });

  try {
    print('Sending request to Gemini...');
    final response = await http.post(
      url, 
      headers: {'Content-Type': 'application/json'}, 
      body: body
    );

    print('Status Code: ${response.statusCode}');
    print('Response Body: ${response.body}');
  } catch (e) {
    print('Exception: $e');
  }
}

File: test_gemini.py
import urllib.request
import json

url = 'https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent?key='
data = json.dumps({"contents":[{"parts":[{"text":"Hello"}]}]}).encode('utf-8')
req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})

try:
    with urllib.request.urlopen(req) as response:
        print(response.read().decode('utf-8'))
except urllib.error.HTTPError as e:
    print(f'HTTP Error {e.code}: {e.reason}')
    print(e.read().decode('utf-8'))
except Exception as e:
    print(f'Error: {e}')
